src/utilities/parsing.ts

```diff
@@ -6,4 +6,12 @@ export function parseBoolean(value: string, defaultValue: boolean): boolean {
     } else {
         return defaultValue;
     }
+}
+
+export function parseUndefinedValue(val: number, initVal: number): number {
```

Comment:
I don't think `parseUndefinedValue` is a helpful name. Perhaps `parseNumber`? Also, wouldn't it be better to use `isfinite` instead of checking the type? For example, if `val` is a `string`, this would return the string instead of a number.